Console/retro/Console.cc

```diff
@@ -265,15 +265,79 @@ void Console::ScrollUp(short n)
     dirtyRect.bottom = dirtyRect.bottom > 0 ? dirtyRect.bottom - 1 : 0;
 }
 
+void Console::ProcessOSCseq(char c)
+{
+    switch(sequenceStep)
+    {
+    case 1:
+        if(c!='0')  // The only recognized sequence is OSC 0;
+        {
+            OSCseq=false;
+            sequenceStep=0;
+            return;
+        }
+        ++sequenceStep;
+        break;
+    case 2:
+        if(c!=';')  // The only recognized sequence is OSC 0;
+        {
+            OSCseq=false;
+            sequenceStep=0;
+            return;
+        }
+        ++sequenceStep;
+        windowName="";
+        break;
+    default:
+        if(c==BEL) // The BEL character ends the sequence.
+        {
+            setWindowName(windowName);
+            OSCseq=false;
+            isProcessingEscSequence=false;
+            sequenceStep=0;
+        }
+        else
+        {
+            windowName+=c;
+            ++sequenceStep;
+        }
+    }
+}
+
 void Console::ProcessEscSequence(char c)
 {
+    if(sequenceStep>0 && OSCseq)
+    {
+        ProcessOSCseq(c);
+        //sequenceStep=0;
+        //isProcessingEscSequence=false;
+        return;
+    }
+    if(sequenceStep>MAX_LEN)
```

Comment:
I'd do this check right next to the `windowName+=c` above, because that's the only thing it really checks for. A plain `if(windowName.size() > ...)` seems clearer than this check.

Reply:
Ok. It fits better to the code organization with the state machine.